match/spec/portal_fetcher_spec.rb

```diff
@@ -0,0 +1,135 @@
+require 'spaceship/client'
+
+describe Match do
+  describe Match::Portal::Fetcher do
+    let(:default_sut) {
+      Match::Portal::Fetcher
+    }
+
+    let(:portal_bundle_id) { double("portal_bundle_id") }
+    let(:portal_device) { double("portal_device") }
+    let(:portal_certificate) { double("portal_certificate") }
+
+    let(:portal_profile) { double("portal_profile") }
+    let(:portal_profile_udid) { double("portal_profile_udid") }
+
+    let(:deviceClass) { Spaceship::ConnectAPI::Device::DeviceClass }
+    let(:deviceStatus) { Spaceship::ConnectAPI::Device::Status }
+    let(:certificateType) { Spaceship::ConnectAPI::Certificate::CertificateType }
+    let(:profileState) { Spaceship::ConnectAPI::Profile::ProfileState }
+
+    let(:default_device_all_params) {
+      {
+        filter: { platform: 'IOS,UNIVERSAL', status: 'ENABLED' },
+        client: anything
+      }
+    }
+
+    let(:default_certificates_all_params) {
+      {
+        filter: { certificateType: anything }
+      }
+    }
+
+    let(:default_profile_all_params) {
+      {
+        filter: { profileType: anything },
+        includes: 'bundleId,devices,certificates'
+      }
+    }
+
+    let(:default_bundle_id_all_params) {
+      {
+        filter: { identifier: anything }
+      }
+    }
+
+    before do
+      allow(portal_device).to receive(:device_class).and_return(deviceClass::IPHONE)
+      allow(portal_device).to receive(:enabled?).and_return(true)
+
+      allow(portal_certificate).to receive(:valid?).and_return(true)
+
+      allow(portal_profile).to receive(:uuid).and_return(portal_profile_udid)
+      allow(portal_profile).to receive(:profile_state).and_return(profileState::ACTIVE)
+      allow(portal_profile).to receive(:devices).and_return([portal_device])
+      allow(portal_profile).to receive(:certificates).and_return([portal_certificate])
+    end
+
+    before do
+      allow(Spaceship::ConnectAPI::Device).to receive(:all).with(default_device_all_params).and_return([portal_device])
+      allow(Spaceship::ConnectAPI::Certificate).to receive(:all).with(default_certificates_all_params).and_return([portal_certificate])
+      allow(Spaceship::ConnectAPI::Profile).to receive(:all).with(default_profile_all_params).and_return([portal_profile])
+      allow(Spaceship::ConnectAPI::BundleId).to receive(:all).with(default_bundle_id_all_params).and_return([portal_bundle_id])
+    end
+
+    describe "certificates" do
+      it "fetches certificates" do
+        # GIVEN
```

Comment:
Nit: I personally appreciate adding "given/when/then" comments to unit tests, but it's not a practice we do in this project, so adding it to these files in this PR would make it inconsistent with the code style of the rest of the project. Do you feel strongly about keeping them?

Reply:
It's just my personal preference. And @\lacostej likes it, too, and even asks me to add it [in another PR](https://github.com/fastlane/fastlane/pull/21691#discussion_r1419585821) 🤣.
Since it nit, I would prefer leave it as it is.